app/server/routes/v1/comment/commentFlags.js

```diff
@@ -0,0 +1,278 @@
+ *                $ref: '#/components/schemas/Comment/properties/flags'
+ *        '400':
+ *          description: Bad Request.
+ */
+
+// Get comment flags by comment id
+router.get("/:id", async (req, res) => {
+  try {
+    req.log.addAction("Finding comment.");
+    const documents = await findSingleDocuments({
+      comment: req.params.id,
+    });
+    req.log.addAction("Comment found.");
+
+    req.log.setResponse(200, "Success", null);
+    return res.status(200).json(documents.comment.flags);
+  } catch (err) {
+    // Explicitly thrown error
+    if (err instanceof ResponseError) {
+      req.log.setResponse(err.status, "ResponseError", err.message);
+      return res.status(err.status).send(err.message);
+    }
+    // Bad Request
+    req.log.setResponse(400, "Error", err);
+    return res.status(400).send(`Bad Request: ${err}`);
+  } finally {
+    req.log.print();
+  }
+});
+
+/**
+ * @swagger
+ * paths:
+ *  /api/comment/flags/{id}:
+ *    post:
+ *      security:
+ *        - bearerAuth: []
+ *      tags:
+ *        - Comment Flags
+ *      summary: Flag comment by comment id.
+ *      parameters:
+ *        - in: path
+ *          required: true
+ *          name: id
+ *          schema:
+ *            $ref: '#/components/schemas/Comment/properties/id'
+ *        - in: query
+ *          required: true
+ *          name: flag
+ *          schema:
+ *            $ref: '#/components/schemas/Comment/properties/flags/items/properties/flag'
+ *      responses:
+ *        '404':
+ *          description: User not found. **||** <br>Comment not found.
+ *        '403':
+ *          description: Invalid flag. Use one of <br>[Inappropriate, Hate, Harassment or Bullying, Spam, Misinformation, Against Community Rules]
+ *        '204':
+ *          description: Success. No content to return.
+ *        '400':
+ *          description: Bad Request.
+ */
+
+// Flag comment by comment id
+router.post("/:id", async (req, res) => {
+  try {
+    req.log.addAction("Finding user and comment.");
+    // TODO: Can documents be done with destructuring?
+    const documents = await findSingleDocuments({
+      user: req.user.username,
+      comment: req.params.id,
+    });
+    req.log.addAction("User and comment found.");
+
+    // TODO: Set flags in an options collection, that can be edited by admins
+    const flags = [
+      "Inappropriate",
+      "Hate",
+      "Harassment or Bullying",
+      "Spam",
+      "Misinformation",
+      "Against Community Rules",
+    ];
+
+    req.log.addAction("Checking flag query is valid.");
+    if (!flags.includes(req.query.flag))
+      throw new ResponseError(
+        403,
+        "Invalid flag. Use one of [Inappropriate, Hate, Harassment or Bullying, Spam, Misinformation, Against Community Rules]"
+      );
+
+    // If flag isn't set on post
+    req.log.addAction("Checking if flag is set on comment.");
+    // TODO: This first part (if statement) may be unnecessary. Second action may just set
```

Comment:
@ZachBourque what's your opinion on this? Commit changes into this branch if required before approving and merging please.

Reply:
@ZachBourque the comment on line 144

Reply:
I think I can test this

Reply:
Tested and it can't be done. Nothing changed.